### PR TITLE
scala3 support for core module

### PR DESCRIPTION
**Pull request by pjfanning**
* migration and projection modules need pekko-persistence-jdbc scala3 support

### CODE REVIEW

File: .github/workflows/build-test.yml [hunk 1]
```diff
@@ -36,6 +36,30 @@ jobs:
...
+          java-version: 8
+
+      - name: Cache Coursier cache
+        uses: coursier/cache-action@v6
+
+      - name: Enable jvm-opts
+        run: cp .jvmopts-ci .jvmopts
+
+      - name: Compile
+        run: sbt ++3.3.0 core/Test/compile
```
Comment (He-Pin) [inline .github/workflows/build-test.yml:61]: `sbt ++3.3.x` https://github.com/sbt/sbt/pull/6936

File: core/src/main/scala/org/apache/pekko/persistence/r2dbc/query/scaladsl/QueryDao.scala [hunk 2]
```diff
@@ -179,10 +179,10 @@ private[r2dbc] class QueryDao(settings: R2dbcSettings, connectionFactory: Connec
       row =>
         if (backtracking)
           SerializedJournalRow(
-            slice = row.get("slice", classOf[Integer]),
+            slice = row.get[Integer]("slice", classOf[Integer]),
```
Comment (He-Pin) [inline core/src/main/scala/org/apache/pekko/persistence/r2dbc/query/scaladsl/QueryDao.scala:182]: Why scala3 can't infer this type info?

**Review by He-Pin — APPROVED** [review]
lgtm

File: core/src/main/scala/org/apache/pekko/persistence/r2dbc/query/scaladsl/QueryDao.scala (hunk 2 above)
Comment (pjfanning) [reply core/src/main/scala/org/apache/pekko/persistence/r2dbc/query/scaladsl/QueryDao.scala:182]: I don't know. Scala 3 is not Scala 2 and does not work excatly the same as Scala 2. The code does not work in Scala 3 if I do not add this change.

Comment (He-Pin) [reply core/src/main/scala/org/apache/pekko/persistence/r2dbc/query/scaladsl/QueryDao.scala:182]: Yes, I was expect Scala3 has better type inference.

**Pull request merged**